python/beeai_framework/tools/tool.py

```diff
@@ -220,12 +220,12 @@ def _create_emitter(self) -> Emitter:
                 creator=self,
             )
 
-        async def _run(self, input: Any, options: ToolRunOptions | None, context: RunContext) -> Any:
+        async def _run(self, input: Any, options: ToolRunOptions | None, context: RunContext) -> StringToolOutput:
```

Comment:
i think the return type hint for the tool _run should probably be ToolOutput especially with the JSONToolOutput is incoming: https://github.com/i-am-bee/beeai-framework/pull/481/files#diff-bbeaddb800a3c89929ddeda85a1c3bfc830376459e30c308c667412197d5ec04R73-R83

Reply:
I was about to say this, I think the tool decorator shouldn't be restricted to just string output

Reply:
We should either use generic `ToolOutput` or ideally introduce a generic for that.